salt/states/git.py

```diff
@@ -721,6 +724,31 @@ def latest(name,
                 )
                 local_changes = False
 
+            if local_changes:
```

Comment:
OK I just pushed a couple commits that make the requested change and update the test.